pkg/util/oom.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/docker/libcontainer/cgroups/fs"
+	"github.com/docker/libcontainer/configs"
+)
+
+const (
+	processListingDelay = 50 * time.Millisecond
+)
+
+// Writes 'value' to /proc/<pid>/oom_score_adj. PID = 0 means self
+func ApplyOomScoreAdj(pid int, value int) error {
+	if value < -1000 || value > 1000 {
+		return fmt.Errorf("invalid value(%d) specified for oom_score_adj. Values must be within the range [-1000, 1000]", value)
+	}
+	if pid < 0 {
+		return fmt.Errorf("invalid PID %d specified for oom_score_adj", pid)
+	}
+
+	var pidStr string
+	if pid == 0 {
+		pidStr = "self"
+	} else {
+		pidStr = strconv.Itoa(pid)
+	}
+
+	oom_value, err := ioutil.ReadFile(path.Join("/proc", pidStr, "oom_score_adj"))
```

Comment:
Consider just setting the OOM score without worrying about the current value. Would simplify the code and not necessarily add overhead.